graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/nodes/cast/JSToStringOrNumberNode.java

```diff
@@ -120,4 +122,14 @@ protected double doUndefined(@SuppressWarnings("unused") Object value) {
     protected static BigInt doBigInt(BigInt value) {
         return value;
     }
+
+    @Specialization
+    protected String doRecord(Record value, @Cached("create()") JSToStringNode toStringNode) {
```

Comment:
I believe you can use `@Shared` here, so that both toStringNodes only use one field in the Node.

Reply:
Done